lib/src/internal/FlowManager.cpp

```diff
@@ -186,22 +208,51 @@ namespace mxl::lib
             // Write the json file to disk.
             writeFlowDescriptor(tempDirectory, flowDef);
 
-            auto flowData = std::make_unique<ContinuousFlowData>(makeFlowDataFilePath(tempDirectory).string().c_str(), AccessMode::CREATE_READ_WRITE);
+            auto flowData = std::unique_ptr<ContinuousFlowData>{};
+            try
+            {
+                flowData = std::make_unique<ContinuousFlowData>(makeFlowDataFilePath(tempDirectory).string().c_str(), AccessMode::CREATE_READ_WRITE);
+            }
+            catch (std::exception const& e)
+            {
+                MXL_ERROR("FlowManager: Failed to create continuous flow data for '{}': {}", uuidString, e.what());
+                throw std::runtime_error(std::string("FlowManager: Failed to mmap continuous flow data: ") + e.what());
+            }
```

Comment:
1. As above wrt. rethrowing the original exception instead of replacing it with something generic.
2. Please use curly braces for initialization of the `std::string`, as well as the exception.